### ISSUE TITLE
Are all AVX-512 SIMD functions and types available in Rust `std::arch` module?

### ISSUE THREAD

**Issue by luojia65** [rust-lang/rust#63769]
Hello! We are developing our game engine (esp. deserialization and noise generation part) in Rust, which theoretically can make good use of modern SIMD features. There could be relevant x86 CPUs available in market with latest AVX-512 instructions, and as we may concern Rust should have full (or partial) support for them in `std::arch` as there are already AVX2 and earlier technologies available. (We have already supported AVX2 etc.) 
Is AVX-512 implementation already on schedule for next Rust nightly, or we should wait by now for upstream updates? Thanks!

---

Edit: some of my friends at rustcc group mentioned if I could use FFI to make an approach, but I do not know if this is relevant to LLVM or other parts as we are new on this.

**Comment by pitdicker** [comment]
I think this is the issue to look :https://github.com/rust-lang/stdarch/issues/310

**Comment by jonas-schievink** [comment]
Yeah, closing in favor of rust-lang/stdarch#310
